backend/src/main/java/com/bulletjournal/messaging/MessagingService.java

```diff
@@ -94,6 +122,42 @@ public void sendEtagUpdateNotificationToUsers(Collection<String> usernames) {
         fcmClient.sendAllMessagesAsync(params);
     }
 
+    public void sendJoinGroupNotificationEmailsToUser(
+        List<Pair<String, Notification>> notificationWithUIDs) {
+        LOGGER.info("Sending join group notifications ...");
+        try {
+            Set<String> distinctTargetUsers = notificationWithUIDs.stream().flatMap(item ->
+                Stream.of(item.getValue().getTargetUser()))
+                .collect(Collectors.toSet());
+            List<User> targetUsers = userDaoJpa.getUsersByNames(distinctTargetUsers);
+            Map<String, String> targetUserEmailMap = new HashMap<>();
+            for (User user : targetUsers) {
+                if (user.getEmail() != null && !user.getEmail().endsWith("@anon.1o24bbs.com")) {
+                    targetUserEmailMap.put(user.getName(), user.getEmail());
+                }
+            }
+
+            Set<String> distinctInviters = notificationWithUIDs.stream().flatMap(item ->
+                Stream.of(item.getValue().getOriginator()))
+                .collect(Collectors.toSet());
+            Map<String, String> inviterAvatarMap = getAvatarMap(new ArrayList<>(distinctInviters));
+
+            List<MailjetEmailParams> emailParamsList = new ArrayList<>();
+            for (Pair<String, Notification> notificationWithUID : notificationWithUIDs) {
+                MailjetEmailParams mailjetEmailParams =
+                    createEmailParamsForGroupInvitation(notificationWithUID,
+                        targetUserEmailMap, inviterAvatarMap);
+                if (mailjetEmailParams != null){
+                    emailParamsList.add(mailjetEmailParams);
+                }
+            }
+            mailjetClient.sendAllEmailAsync(emailParamsList);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
```

Comment:
`LOGGER.error("sendJoinGroupNotificationEmailsToUser failed", e);`

So that there is stacktrace

Reply:
Updated